instrumentation/shoryuken/lib/opentelemetry/instrumentation/shoryuken/middlewares/server/tracer_middleware.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+module OpenTelemetry
+  module Instrumentation
+    module Shoryuken
+      module Middlewares
+        module Server
+          # TracerMiddleware propagates context and instruments Shoryuken requests
+          # by way of its middleware system
+          class TracerMiddleware
+            def call(worker_instance, queue, sqs_msg, _body)
+              attributes = {
+                OpenTelemetry::SemanticConventions::Trace::MESSAGING_SYSTEM => 'shoryuken',
+                'code.namespace' => worker_instance.class.name,
+                OpenTelemetry::SemanticConventions::Trace::MESSAGING_MESSAGE_ID => sqs_msg.message_id,
+                OpenTelemetry::SemanticConventions::Trace::MESSAGING_DESTINATION => queue,
+                OpenTelemetry::SemanticConventions::Trace::MESSAGING_DESTINATION_KIND => 'queue',
+                OpenTelemetry::SemanticConventions::Trace::MESSAGING_OPERATION => 'process'
+              }
+              span_name = case instrumentation_config[:span_naming]
+                          when :job_class then "#{worker_instance.class.name} process"
+                          else "#{queue} process"
+                          end
+
+              extracted_context = OpenTelemetry.propagation.extract(sqs_msg)
+              OpenTelemetry::Context.with_current(extracted_context) do
+                links = []
+                span_context = OpenTelemetry::Trace.current_span(extracted_context).context
+                links << OpenTelemetry::Trace::Link.new(span_context) if span_context.valid?
+                span = tracer.start_root_span(span_name, attributes: attributes, links: links, kind: :consumer)
+                OpenTelemetry::Trace.with_span(span) do
+                  yield
+                rescue Exception => e # rubocop:disable Lint/RescueException
+                  span&.record_exception(e)
```

Comment:
Any reason to use `span&` in this and following lines?  `span` is assigned before this block. Is there any chance `span = tracer.start_root_span(span_name, attributes: attributes, links: links, kind: :consumer)` returns `nil`?

Reply:
That is a good point since the exception handling is _only_ in the with context block. In that case `span` will always be set.

Reply:
Happy to go either way on this. I had updated it on @arielvalentin's [suggestion](https://github.com/open-telemetry/opentelemetry-ruby-contrib/pull/536#discussion_r1248319650). Not sure if there is much chance of `span` being `nil` in this context, but don't see a downside to using the `span&` here.

Reply:
@mcinbell anyway this is not blocker as mentioned, I can open small PR once this is merged to address this (with full reasoning).

Reply:
That's fine. As I stated, my original implementation was `span.record_exception` so am happy to revert the change if @arielvalentin is happy with that. Personally, I don't imagine the `span&` adds much confusion but agree that it's better not to use it if it's unnecessary. As I have stated in other comments, this implementation borrowed heavily from the sidekiq instrumentation which did not use the `span&` in the equivalent location, however am happy to be guided by the reviewers on this point.

Reply:
Apologies for the confusion. When I first reviewed this I thought the exception handler was also wrapping the outer block where the span was being created and now realize that it was only scoped to handle exceptions raised from the yield. 

Thanks again for your patience and I'm ok with removing safe navigation.